be/src/exec/stream/state/imt_state_table.h

```diff
@@ -0,0 +1,74 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include "column/datum_tuple.h"
+#include "column/vectorized_field.h"
+#include "column/vectorized_schema.h"
+#include "exec/stream/state/state_table.h"
+#include "exec/tablet_sink.h"
+#include "gen_cpp/Descriptors_types.h"
+#include "storage/chunk_iterator.h"
+#include "storage/table_reader.h"
+
+namespace starrocks::stream {
+
+using DatumKeyRow = std::vector<DatumKey>;
+
+class IMTStateTable : public StateTable {
```

Comment:
Should it be reused among epoch to reduce the construction overhead ?

Reply:
Yep. `IMTStateTable ` is prepared in `StreamAggregaor` in `open` stage.

However if `TableReader` need refresh version for each Epoch, maybe need reset it in `reset_epoch`.